### PR TITLE
issue: 1679157 Fix gcc8 build warnings

### PR DESCRIPTION
**Pull request by liranoz12**
Signed-off-by: Liran Oz <lirano@mellanox.com>

### CODE REVIEW

File: src/message.cpp [hunk 1]
```diff
@@ -93,7 +93,7 @@ Message::Message() {
 
     for (int len = 0; len < ms_maxSize; len++)
         m_addr[len] = (uint8_t)rand();
-    memset(m_header, 0, MsgHeader::EFFECTIVE_SIZE);
+    memset((char*)m_header, 0, MsgHeader::EFFECTIVE_SIZE);
```
Comment (igor-ivanov) [inline src/message.cpp]: why it is not `(void *)`?

File: src/sockperf.cpp [hunk 2]
```diff
@@ -2156,7 +2156,7 @@ void set_defaults() {
     int igmp_max_memberships = read_int_from_sys_file("/proc/sys/net/ipv4/igmp_max_memberships");
     if (igmp_max_memberships != -1) IGMP_MAX_MEMBERSHIPS = igmp_max_memberships;
 
-    memset(&s_user_params, 0, sizeof(struct user_params_t));
+    memset((char*)&s_user_params, 0, sizeof(s_user_params));
```
Comment (igor-ivanov) [inline src/sockperf.cpp]: why it is not `(void *)`?

**Comment by liranoz12** [comment]
@igor-ivanov see update

**Review by igor-ivanov — APPROVED** [review]
👍

**Pull request merged**